crates/core/tedge_agent/src/file_transfer_server/http_rest.rs

```diff
@@ -59,18 +59,18 @@ async fn upload_file(
     }
 }
 
-fn source_err_is_is_a_directory(error: &anyhow::Error) -> bool {
+fn source_err_is_is_a_directory(error: &anyhow::Error, path: &Utf8Path) -> bool {
     error
         .downcast_ref()
-        .map(err_is_is_a_directory)
+        .map(|e| err_is_is_a_directory(e, path))
         .unwrap_or(false)
 }
 
 async fn download_file(
     path: FileTransferPath,
 ) -> Result<StreamBody<ReaderStream<BufReader<File>>>, Error> {
     let reader: Result<_, io::Error> = async {
-        let mut buf_reader = BufReader::new(File::open(path.full).await?);
+        let mut buf_reader = BufReader::new(File::open(path.full.to_owned()).await?);
```

Comment:
Passing a reference was sufficient:
```suggestion
        let mut buf_reader = BufReader::new(File::open(&path.full).await?);
```

Reply:
Good catch. I just pushed the change (and squashed the fixup already since it was only one minor change).